dist/cms.js

```diff
@@ -77,11 +77,86 @@ function getWordPressScripts() {
   return entries;
 }
 
+/**
+ * Detects the type of WordPress content for the current document.
+ *
+ * @returns {object} Object with fields `template`, `post_type`, and `taxonomy`.
+ */
+function getWordPressContentType() {
+  const content = {
+    template: 'unknown',
+    post_type: '',
+    taxonomy: '',
+  };
+  try {
+    const bodyClass = document.body.classList;
+
+    if ( bodyClass.contains( 'home' ) ) {
+      /*
+       * The home page, either containing the blog,
+       * or a "static front page".
+       */
+      if ( bodyClass.contains( 'blog' ) ) {
+        content.template = 'home-blog';
+        content.post_type = 'post';
+      } else if ( bodyClass.contains( 'page' ) ) {
+        content.template = 'home-page';
+        content.post_type = 'page';
+      }
+    } else if ( bodyClass.contains( 'blog' ) ) {
```

Comment:
Probably a coding style preference here but there's a lot of `else if`'s in this, including nested ones, with no real common code other than the return statement.

Should we refactor to just return early to make this easier to follow?

For example:

```
    if ( bodyClass.contains( 'home' ) ) {
      /*
       * The home page, either containing the blog,
       * or a "static front page".
       */
      if ( bodyClass.contains( 'blog' ) ) {
        content.template = 'home-blog';
        content.post_type = 'post';
        return content;
      }
 
      if ( bodyClass.contains( 'page' ) ) {
        content.template = 'home-page';
        content.post_type = 'page';
        return content;
      }

      return content;
    }
    
    if ( bodyClass.contains( 'blog' ) ) {
```

Or alternatively could go for a much flatter option, but this does duplicate some code:

```
    if ( bodyClass.contains( 'home' ) && bodyClass.contains( 'blog' ) ) {
      content.template = 'home-blog';
      content.post_type = 'post';
      return content;
    }
 
    if ( bodyClass.contains( 'home' ) && bodyClass.contains( 'page' ) ) {
      content.template = 'home-page';
      content.post_type = 'page';
      return content;
    }
    
    if ( bodyClass.contains( 'blog' ) ) {
```

Reply:
IMO, the addition of many more `return` statements makes the code more complex. I like to have just a single `return` when possible.

Reply:
I’m definitely a fan of early return options where possible but agree this case it’s not exactly that. Though potentially it should early return if bodyClass is empty rather than waste time going through all the cases.

Thinking about the third option also potentially surfaced a flaw in the logic. If the body class contains `home` but not `blog` nor `page` do we really wanna return the default `unknown` template? Or should it try the other checks to see if it can get a better match?

Reply:
@tunetheweb Realistically, there would never be a situation where `home` is applied without `blog` or `page`, simply because WordPress's functionality dictates that the home page is either the blog or a page. We could support returning a plain `home` value (instead of `home-blog` and `home-page`), but that wouldn't really provide value I think. So I think returning `unknown` makes more sense as this is not something that WordPress supports out of the box. If there was a `home` class only, we would know that a plugin has tampered with those classes in which case the detection isn't reliable.